src/egraph.rs

```diff
@@ -46,10 +46,10 @@ and
 pub struct EGraph<L: Language, N: Analysis<L>> {
     /// The `Analysis` given when creating this `EGraph`.
     pub analysis: N,
+    pub explain: Explain<L>,
+    to_union: Vec<(Id, Id, Option<String>)>,
```

Comment:
This is the new to_union stack. When proof production is on, it is important that these id's are not canonical- they represent the enode-ids of the two enodes that are the match and application.